manifests/net_process_log.pp

```diff
@@ -0,0 +1,73 @@
+# @summary Logs user processes and open network connections to syslog
+#
+# Logs user processes and open network connections to syslog
+#
+# @param enable_net_process_log
+#   Enable/disable the net_process_log logging
+#
+# @param ignore_users
+#   String to pass to net_process_log.pl script as -i argument.
+#   Setting to empty string will use default defined in net_process_log.pl.
+#   Format is a csv of users to ignore. Example : "root,chrony,dbus".
+#
+# @param ps_arg
+#   String to pass to net_process_log.pl script as --psarg argument.
+#   Setting to empty string will use default defined in net_process_log.pl.
+#
+# @param ss_arg
+#   String to pass to net_process_log.pl script as --ssarg argument.
+#   Setting to empty string will use default defined in net_process_log.pl
+#   
+# @param ss_filter
+#   String to pass to net_process_log.pl script as --ssfilter argument.
+#   Setting to empty string will use default defined in net_process_log.pl
+#
+# @param minute_interval
+#   Value to use in the minute field of the cron task 
+#
+# @example
+#   include profile_audit::net_process_log
+class profile_audit::net_process_log (
+  Boolean $enable_net_process_log,
+  String $ignore_users,
+  String $ps_arg,
+  String $ss_arg,
+  String $ss_filter,
+  String $minute_interval,
+) {
+
+  if ($enable_net_process_log) {
+    $ensure_parm = 'present'
+  } else {
+    $ensure_parm = 'absent'
+  }
+
+  file { '/root/scripts/net_process_log.pl':
+    ensure => $ensure_parm,
+    mode   => '0750',
+    owner  => 'root',
+    group  => 'root',
+    source => "puppet:///modules/${module_name}/net_process_log.pl",
+    require => File["/root/scripts"],
+  }
```

Comment:
One small issue. We have been trying to move scripts that are run via CRON into `/root/cron_scripts` rather than just `/root/scripts`.

I cannot remember the details of why, but Andy suggested this in a review of another profile, and it made sense to me at that time. We haven't moved all scripts run from CRON into this location yet, but we should try to do that for anything new like this.

Reply:
```suggestion
  file { '/root/cron_scripts/net_process_log.pl':
    ensure => $ensure_parm,
    mode   => '0750',
    owner  => 'root',
    group  => 'root',
    source => "puppet:///modules/${module_name}/net_process_log.pl",
    require => File["/root/cron_scripts"],
  }
```